src/MongoDB/Driver/Server.cpp

```diff
@@ -78,39 +90,51 @@ Object hippo_mongo_driver_server_create_from_id(mongoc_client_t *client, uint32_
 	return tmp;
 }
 
+hippo_server_description_type_t hippo_server_description_type(mongoc_server_description_t *sd)
+{
+	int i;
+	const char *name = mongoc_server_description_type(sd);
+
+	for (i = 0; i < HIPPO_SERVER_DESCRIPTION_TYPES; i++) {
+		if (strcmp(name, hippo_server_description_type_map[i].name) == 0) {
+			return hippo_server_description_type_map[i].type;
+		}
+	}
+
+	return HIPPO_SERVER_UNKNOWN;
+}
+
 
 static bool mongodb_driver_add_server_debug(mongoc_server_description_t *sd, Array *retval)
 {
+	mongoc_host_list_t *host = mongoc_server_description_host(sd);
+	const bson_t       *is_master = mongoc_server_description_ismaster(sd);
+
 	Variant v_last_is_master;
 	Array   a_last_is_master;
 
-	retval->set(s_MongoDriverServer_host, sd->host.host);
-	retval->set(s_MongoDriverServer_port, sd->host.port);
-	retval->set(s_MongoDriverServer_type, sd->type);
-	retval->set(s_MongoDriverServer_is_primary, !!(sd->type == MONGOC_SERVER_RS_PRIMARY));
-	retval->set(s_MongoDriverServer_is_secondary, !!(sd->type == MONGOC_SERVER_RS_SECONDARY));
-	retval->set(s_MongoDriverServer_is_arbiter, !!(sd->type == MONGOC_SERVER_RS_ARBITER));
+	retval->set(s_MongoDriverServer_host, host->host);
+	retval->set(s_MongoDriverServer_port, host->port);
+	retval->set(s_MongoDriverServer_type, hippo_server_description_type(sd));
+	retval->set(s_MongoDriverServer_is_primary, strcmp(mongoc_server_description_type(sd), hippo_server_description_type_map[HIPPO_SERVER_RS_PRIMARY].name) == 0);
+	retval->set(s_MongoDriverServer_is_secondary, strcmp(mongoc_server_description_type(sd), hippo_server_description_type_map[HIPPO_SERVER_RS_SECONDARY].name) == 0);
+	retval->set(s_MongoDriverServer_is_arbiter, strcmp(mongoc_server_description_type(sd), hippo_server_description_type_map[HIPPO_SERVER_RS_ARBITER].name) == 0);
```

Comment:
I believe you want `!!` here or some other explicit cast to a boolean. As-is, you're prone to reporting -1, 0, or 1 via these three fields.